spring-transaction/build.gradle.kts

```diff
@@ -27,6 +27,7 @@ dependencies {
     testImplementation("org.apache.logging.log4j", "log4j-api", Versions.log4j2)
     testImplementation("org.apache.logging.log4j", "log4j-core", Versions.log4j2)
     testImplementation("junit", "junit", "4.12")
+    testImplementation("io.mockk", "mockk", "1.13.7")
```

Comment:
We would prefer to avoid requiring a `mockk` dependency for the new test file. Ideally, each unit test in `SpringTransactionManagerTest.kt` would check the specific test case using properties or returned values, instead of checking that required methods were invoked.

I understand that it requires more code set-up on your part, but if the goal of each test is just to confirm transaction connection calls without performing any transaction operations that can return results, you can reference how it's been done in [ThreadLocalManagerTest.kt](https://github.com/JetBrains/Exposed/blob/main/exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ThreadLocalManagerTest.kt#L103). On my end, it just requires adding 2 base classes:
```kt
// kept open so you can add any necessary subclasses that mock throwing exceptions in the functions
private open class ConnectionSpy(private val connection: Connection) : Connection by connection {
    var commitCallCount = 0
    var rollbackCallCount = 0
    var closeCallCount = 0
    // if function call order really needs to be tested, calls can be stored in a collection that preserves order too

    override fun commit() { commitCallCount++ }
    override fun rollback() { rollbackCallCount++ }
    override fun close() { closeCallCount++ }
}

private class MockedDataSource<T : Connection>(private val connectionDecorator: (Connection) -> T) : DataSource {
    val connections = mutableListOf<T>()

    override fun getConnection(): Connection {
        val connection = DriverManager.getConnection(TestDB.H2.connection(), TestDB.H2.user, TestDB.H2.pass)
        val wrapped = connectionDecorator(connection)
        connections.add(wrapped)
        return wrapped
    } 
    // all other necessary overrides throw NotImplementedError()
}

// which could then used like so:
private val ds1 by lazy { MockedDataSource(::ConnectionSpy) }
// and then in tests like so:
assertEquals(1, ds1.connections.size)
ds1.connections.forEach {
    assertEquals(1, it.commitCallCount)
    assertEquals(1, it.closeCallCount)
}
```

Reply:
I introduced mockk because there is too much code in the method and it is difficult to check the order in which it is called. 
Your kind explanation made me realize the Exposed team's philosophy. I will change the code tomorrow and write tests in the way you suggested.